core/iwasm/common/gc/gc_common.c

```diff
@@ -196,16 +331,48 @@ bool
 wasm_obj_is_instance_of_defined_type(WASMObjectRef obj, WASMType *defined_type,
                                      WASMModuleCommon *const module)
 {
-    /* TODO */
-    return false;
+    WASMType **types = NULL;
+    uint32 type_count = 0;
+    uint32 type_idx = 0;
+
+#if WASM_ENABLE_INTERP != 0
+    if (module->module_type == Wasm_Module_Bytecode) {
+        WASMModule *wasm_module = (WASMModule*)module;
+        type_count = wasm_module->type_count;
+        types = wasm_module->types;
+    }
+#endif
+
+#if WASM_ENABLE_AOT != 0
+    // TODO
+#endif
+
+    for ( ; type_idx < type_count; type_idx++) {
+        if (wasm_defined_type_equal(types[type_idx], defined_type, module)) {
```

Comment:
Use `if (types[type_idx] == defined_type)`
Or add a common field type_idx for WASMTYPE, WASMFuncType, WASMStructType, WASMArrayType, init it in the loader, and here we get it directly.